var/docs/docker.md

```diff
@@ -0,0 +1,66 @@
+# Docker
+
+We bundle a simple docker that can be used for test purposes.  Either pull it from docker hub or build it.
+
+## Build the docker:
+
+    docker build -t kimai/kimai2 .
+
+## Run the docker
+
+    docker run -ti -p 8001:8001 --name kimai2 --rm kimai/kimai2
+
+The you can then access the site on http://127.0.0.1:8001 or on Mac you may need to use the docker IP:
+
+    docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' kimai2
+
+You can hit kimai at that ip on port 8001.
+
+## Running commands in the docker
+
+You can run any command in the container in this fashion once it is started.  Add -ti to attach a terminal.
+
+    docker exec -ti kimai2 bash
+
+## Create a user and dummy data
+
+See the docs [here](installation.md) for full instructions, but this crates a user admin/admin with all privileges.
+
+    docker exec kimai2 bin/console kimai:create-user admin admin@example.com ROLE_SUPER_ADMIN admin
+
+To install the fixtures:
+
+    docker exec kimai2 bin/console kimai:reset-dev
+
+## Developing against the docker
+
+The following installs assume you have cloned the repo and opened a terminal in the root of the project.
+
+### Grab an initialized sqlite database
+
+If you already have a sqlite DB set up you can skip this step.
+
+    docker run -v /tmp:/tmp kimai/kimai2 cp /opt/kimai/var/data/kimai.sqlite /tmp
```

Comment:
I tried that, but wasn't able to use an already existing sqlite. Yes, this again is duplicating docker docu, but I'd like to explain clearly how it works. So assuming I have my existing SQLITE db file at `/home/kimai/kimai2/data.sqlite` what do I need to execute to use it in my docker image?
I am pretty sure: if we don't explain that now, there will be support requests... ;-)

Reply:
I've changed the documentation to try and better explain what is happening.

Reply:
Yes: noob question, sorry again ^^ 
But what am I doing wrong here? Inside the container:
```
docker exec -ti kimai2 bash
www-data@48ba19368a12:/opt/kimai$ ls -la /tmp/
total 8
drwxrwxrwt  2 root root 4096 Oct 28 21:17 .
drwxr-xr-x 76 root root 4096 Oct 29 13:41 ..
www-data@48ba19368a12:/opt/kimai$ ls -la var/data/
total 860
drwxr-xr-x 2 www-data www-data   4096 Oct 28 21:17 .
drwxr-xr-x 8 www-data www-data   4096 Oct 29 13:41 ..
-rw-r--r-- 1 www-data www-data      0 Oct 28 21:17 .gitkeep
-rw-r--r-- 1 www-data www-data 102400 Oct 28 21:17 kimai.sqlite
-rw-r--r-- 1 www-data www-data 770048 Oct 28 21:17 kimai_test.sqlite
```
Outside the container:
```
ls -la var/data
total 8416
drwxr-xr-x   7 kevin  staff      224 29 Okt 14:40 .
drwxr-xr-x  10 kevin  staff      320 18 Sep 00:20 ..
-rw-r--r--   1 kevin  staff        0 12 Jan  2018 .gitkeep
-rw-r--r--   1 kevin  staff  1323008  2 Okt 21:43 kimai.sqlite
-rw-r--r--   1 kevin  staff   999424 25 Okt 23:48 kimai2.sqlite
-rw-r--r--   1 kevin  staff   266240  8 Sep 10:51 kimai_import.sqlite
-rw-r--r--   1 kevin  staff   770048 21 Sep 21:48 kimai_test.sqlite
```
Totally different files, so simply copying the SQLite file to the location outside the container won't affect Kimai inside the container (do I need a re-built in between?).

And I still don't get what that command is supposed to do: `docker run -v /tmp:/tmp kimai/kimai2 cp /opt/kimai/var/data/kimai.sqlite /tmp`
I tried it, but I couldn't see any changes inside and outside the box.

Reply:
I posted just before you committed ;-) Okay, I guess I do understand now what should happen, but it doesn't:
```
ls -la var/data
total 8416
drwxr-xr-x   7 kevin  staff      224 29 Okt 14:40 .
drwxr-xr-x  10 kevin  staff      320 18 Sep 00:20 ..
-rw-r--r--   1 kevin  staff        0 12 Jan  2018 .gitkeep
-rw-r--r--   1 kevin  staff  1323008  2 Okt 21:43 kimai.sqlite
-rw-r--r--   1 kevin  staff   999424 25 Okt 23:48 kimai2.sqlite
-rw-r--r--   1 kevin  staff   266240  8 Sep 10:51 kimai_import.sqlite
-rw-r--r--   1 kevin  staff   770048 21 Sep 21:48 kimai_test.sqlite

docker run -v /tmp:/tmp kevin/kimai2 cp var/data/kimai.sqlite /tmp

docker exec -ti kimai2 bash
www-data@48ba19368a12:/opt/kimai$ ls -la /tmp/
total 8
drwxrwxrwt  2 root root 4096 Oct 28 21:17 .
drwxr-xr-x 76 root root 4096 Oct 29 13:41 ..
www-data@48ba19368a12:/opt/kimai$ exit
exit

ls -la /tmp/
total 0
drwxrwxrwt    6 root   wheel   192 29 Okt 14:41 .
drwxr-xr-x    6 root   wheel   192  3 Okt  2017 ..
drwx------    4 kevin  wheel   128 28 Okt 22:30 .vbox-kevin-ipc
drwx------    3 kevin  wheel    96 17 Okt 07:55 com.apple.launchd.dNTaaolgeB
drwx------    3 kevin  wheel    96 17 Okt 07:55 com.apple.launchd.pyPDOHwTwk
drwxr-xr-x  287 kevin  wheel  9184 24 Okt 10:25 github-changelog-http-cache
```
Neither in the container nor on my host is the kimai.sqlite file in /tmp/ ?

Reply:
I see now the gaps in my docs.

When you build or pull the image it creates an initialised sqlite file.  This has the schema and migrations run in it.  The run command starts a new instance image (a container) and copies the initialised DB to /tmp.  Because we had mount our tmp into the container when the container exists the db is still available on our host system. We then copy that into the project on the host system.

When we run the image as a developer we mount our local project folder over the one shipped in the image.  The sqlite db that we copied out is now available to the project using the php stack in the container.

Equally you should be able to pick any sqlite db from another kimai2 and copy it to var/data/kimai.sql and when you start the container it will be used.  This is all true as long as the local project id mounted over the one shipped in the container.

It all relies on the ```-v $(pwd):/opt/kimai``` to mount the local folder into the container at run time.

Reply:
That's wierd:

```bash
tobias@tobias:kimai2 (docker=) $ ls /tmp/*.sqlite
ls: cannot access '/tmp/*.sqlite': No such file or directory
tobias@tobias:kimai2 (docker=) $ docker run -v /tmp:/tmp kimai/kimai2 cp /opt/kimai/var/data/kimai.sqlite /tmp
tobias@tobias:kimai2 (docker=) $ ls /tmp/*.sqlite
/tmp/kimai.sqlite
tobias@tobias:kimai2 (docker=) $ 
```

Reply:
All those command are on my host machine, I'm never executing into an existing container.

I think you may have a running container. Check with a ```docker ps -a```

You may want ```docker system prune`` once you've stopped any old containers.  That cleans up stopped containers.

Reply:
Alright, I'll wait for your next ping